SpatialGDK/Source/SpatialGDK/Private/Schema/SkeletonEntityManifest.cpp

```diff
@@ -0,0 +1,46 @@
+﻿#include "Schema/SkeletonEntityManifest.h"
+
+namespace SpatialGDK
+{
+static void ApplyCollection(TArray<Worker_EntityId>& Array, const Schema_Object& ComponentObject, Schema_FieldId FieldId)
+{
+	const uint32 EntitiesToPopulateCount = Schema_GetEntityIdCount(&ComponentObject, FieldId);
+	Array.SetNumUninitialized(EntitiesToPopulateCount);
+	for (uint32 EntityToPopulateIndex = 0; EntityToPopulateIndex < EntitiesToPopulateCount; ++EntityToPopulateIndex)
+	{
+		Array[EntityToPopulateIndex] = Schema_IndexEntityId(&ComponentObject, FieldId, EntityToPopulateIndex);
+	}
+}
+
+static void ApplyCollection(TSet<Worker_EntityId_Key>& Set, const Schema_Object& ComponentObject, Schema_FieldId FieldId)
+{
+	const uint32 EntitiesToPopulateCount = Schema_GetEntityIdCount(&ComponentObject, FieldId);
+	for (uint32 EntityToPopulateIndex = 0; EntityToPopulateIndex < EntitiesToPopulateCount; ++EntityToPopulateIndex)
+	{
+		const Worker_EntityId EntityId = Schema_IndexEntityId(&ComponentObject, FieldId, EntityToPopulateIndex);
+		Set.Emplace(EntityId);
+	}
+}
+
+void FSkeletonEntityManifest::ApplySchema(const Schema_Object& ComponentObject)
+{
+	ApplyCollection(EntitiesToPopulate, ComponentObject, SpatialConstants::SKELETON_ENTITY_MANIFEST_ENTITIES_TO_POPULATE_ID);
+	ApplyCollection(PopulatedEntities, ComponentObject, SpatialConstants::SKELETON_ENTITY_MANIFEST_POPULATED_SKELETON_ENTITIES_ID);
+}
+
+template <class TCollection>
+void WriteCollection(const TCollection& EntityIds, Schema_Object& ComponentObject, Schema_FieldId FieldId)
+{
+	for (const auto CollectionEntityId : EntityIds)
+	{
+		const Worker_EntityId EntityId = CollectionEntityId;
+		Schema_AddEntityId(&ComponentObject, FieldId, EntityId);
```

Comment:
Generally this is probably better to do by allocating a list on the schema object and then copying to there. If we want to make generic tools to do this they probably shouldn't be as part of a specific components utils.

Reply:
I'm not sure this is as clearly cut:

* Need to allocate bytes via `Schema_AllocateBuffer` - in this case, we'd need to allocate `sizeof(Worker_EntityId) * EntityIds.Num()` bytes
* And we need to iterate over the values from the `EntityIds` TSet<> anyway. 

I'd prefer not to do this memory juggling in this PR, and preferably with generic tools.

Reply:
Just thought that your comment might've been "why make `WriteCollection` a template" instead of "we'd better use `AddEntityIdList` in our generic tools"; `WriteCollection` is a template in this case due to me considering `TArray<EntityId>` over `TSet<>`, and I've made `WriteCollection` to work with either.